src/rpc/blockchain.cpp

```diff
@@ -1623,6 +1623,7 @@ static RPCHelpMan getchaintxstats()
     }
 
     const CBlockIndex* pindexPast = pindex->GetAncestor(pindex->nHeight - blockcount);
+    CHECK_NONFATAL(pindexPast != nullptr);
```

Comment:
Unrelated: Could probably make sense to convert `CHECK_NONFATAL` to the identity function like `Assert`

Reply:
I'd be happy to work on this

Reply:
Since #24812 got merged, I used the identity CHECK_NONFATAL macro in this PR as well.